tests/it/timeout.rs

```diff
@@ -0,0 +1,87 @@
+use std::time::Duration;
+
+use xshell::{cmd, Shell};
+
+#[test]
+fn test_run_timeout_success() {
+    let sh = Shell::new().unwrap();
+    let command = cmd!(sh, "sleep 1"); // Command that sleeps for 1 second
+
+    // Run the command with a timeout of 3 seconds
+    let result = command.run_timeout(Duration::from_secs(3));
+    assert!(result.is_ok(), "Command should complete successfully within the timeout");
+}
+
+#[test]
+fn test_run_timeout_failure() {
+    let sh = Shell::new().unwrap();
+    let command = cmd!(sh, "sleep 5"); // Command that sleeps for 5 seconds
+
+    // Run the command with a timeout of 3 seconds
+    let result = command.run_timeout(Duration::from_secs(3));
+    assert!(result.is_err(), "Command should fail due to timeout");
+}
+
+#[test]
+fn test_read_timeout_success() {
+    let sh = Shell::new().unwrap();
+    let command = cmd!(sh, "echo Hello, world!"); // Command that prints a message
+
+    // Run the command with a timeout of 3 seconds and read stdout
+    let result = command.read_timeout(Duration::from_secs(3));
+    assert!(result.is_ok(), "Command should complete successfully within the timeout");
+    assert_eq!(result.unwrap(), "Hello, world!");
+}
+
+#[test]
+fn test_read_timeout_failure() {
+    let sh = Shell::new().unwrap();
+    let command = cmd!(sh, "sleep 5"); // Command that sleeps for 5 seconds
+
+    // Run the command with a timeout of 3 seconds and read stdout
+    let result = command.read_timeout(Duration::from_secs(3));
+    assert!(result.is_err(), "Command should fail due to timeout");
+}
+
+#[test]
+fn test_read_stderr_timeout_success() {
+    let sh = Shell::new().unwrap();
+    let command = cmd!(sh, "sh -c 'echo Error message 1>&2'"); // Command that prints an error message to stderr
+
+    // Run the command with a timeout of 3 seconds and read stderr
+    let result = command.read_stderr_timeout(Duration::from_secs(3));
+    assert!(result.is_ok(), "Command should complete successfully within the timeout");
+    assert_eq!(result.unwrap(), "Error message");
+}
+
+#[test]
+fn test_read_stderr_timeout_failure() {
+    let sh = Shell::new().unwrap();
+    let command = cmd!(sh, "sleep 5"); // Command that sleeps for 5 seconds
```

Comment:
`sleep` here is an utility from the OS, which is OS-specific and might not be present. For tests, we would want to use our own version of `sleep`. See how the `xecho` is implemented and used for similar tests, and use the same pattern for `xsleep`.